src/test/java/teammates/test/cases/action/InstructorCourseDeleteAllRecoveryCoursesActionTest.java

```diff
@@ -51,12 +52,43 @@ public void testExecuteAndPostProcess() throws Exception {
         assertEquals(instructor2OfCourse3.courseId, courseList.get(0).getId());
         assertTrue(CoursesLogic.inst().isCoursePresent(instructor2OfCourse3.courseId));
 
-        ______TS("Typical case, delete all courses from Recycle Bin, with privilege");
+        ______TS("Typical case, delete all courses from Recycling Bin, with privilege for only some courses");
 
         InstructorAttributes instructor1OfCourse3 = typicalBundle.instructors.get("instructor1OfCourse3");
+        InstructorAttributes newInstructor = InstructorAttributes
+                .builder(instructor1OfCourse3.getGoogleId(), "icdat.owncourse", "Instructor1 Course3", "instructor1@course3.tmt")
+                .build();
+        gaeSimulation.loginAsInstructor(instructor1OfCourse3.googleId);
+        instructorList.remove(instructor2OfCourse3);
+        instructorList.add(instructor1OfCourse3);
+        instructorList.add(newInstructor);
+        courseList = CoursesLogic.inst().getCoursesForInstructor(instructorList);
+        assertEquals(1, courseList.size());
+        CoursesLogic.inst().moveCourseToRecovery("icdat.owncourse");
+        courseList = CoursesLogic.inst().getRecoveryCoursesForInstructors(instructorList);
+        assertEquals(2, courseList.size());
```

Comment:
Why is it necessary to have `instructor1OfCourse3` in the `instructorList` as well?

Reply:
Because there are totally 2 courses under this instructor account (`typicalCourse3` & `icdat.owncourse`), thus got 2 instructor attributes :)